code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -702,7 +697,8 @@
 	if(!istype(apc))
 		to_chat(owner, span_notice("You are already in your Main Core."))
 		return
-	if(SEND_SIGNAL(owner, COMSIG_SILICON_AI_CORE_STATUS) & COMPONENT_CORE_ALL_GOOD)
+	var/mob/living/silicon/ai/ai_owner = astype(owner, /mob/living/silicon/ai)
+	if(ai_owner?.linked_core)
```

Comment:
`if(astype(owner, /mob/living/silicon/ai)?.linked_core)`? since we're not using the var after